Implement daemon mode

andig opened the pull request:
Some code is borrowed from https://github.com/phlib/console-process (LGPL)- need to double-check if that is ok

szepeviktor commented:
Any word about it in the README?

marcj commented:
We can't use code from a different license if it's just simply copied. If it's inspired or is a construct that is used also everywhere else (like in the documentation, tutorials etc) then we don't need the reference to phlib/console-process as those stuff is not licensable. I personally don't see anything that is defendable although I haven't checked the origin and don't know how hard you copied, if you even did.

andig commented:
> We can't use code from a different license if it's just simply copied.

Imho they could chose to relicense for us. I've opened an issue at that project. If you have objections please feel free to close but I'm lacking capacity to start over :/

chrisminett commented:
Hi,

I'm a maintainer at phlib/console-process

It's fine with us! See comment at https://github.com/phlib/console-process/issues/4

Cheers,
ChrisM

andig commented:
I've updated and simplified the PR. I would appreciate a second look as the PHP 7 build fails with an unclear error message. Not sure if this is intermittent or real?

marcj commented:
What is the main processes dies? Will it automatically restart?

andig commented:
No, thats nothing I‘ve considered. Should be done via systemctrl daemon setting?

mowangjuanzi commented:
When will it be merged to master? I need it

acasademont commented:
Hi @andig should we move forward with this one?

jkrzefski commented on src/Commands/StartCommand.php:
```diff
@@ -30,6 +32,20 @@ protected function configure()
+                if ($output->isVerbose()) {
+                    $output->writeln('Daemonized');
+                }
+                return null;
```
As [this project supports Symfony 5](https://github.com/php-pm/php-pm/blob/2.1.1/composer.json#L7), the return value of a commands `execute` method must be an integer. While it is [not part of the methods signature](https://github.com/symfony/console/blob/v5.1.3/Command/Command.php#L162) in the abstract `Command` class, it is still [validated and an exception will be thrown](https://github.com/symfony/console/blob/v5.1.3/Command/Command.php#L260) for any non-integer return value. So I suggest, you change this to `return 0;` along with the other return statement below.

andig commented:
> Hi @andig should we move forward with this one?

Up to you. We‘ve had this around for so long I wouldn‘t mind closing although there seems to be some positive comments.
